js_modules/dagster-ui/packages/ui-core/src/asset-selection/AntlrAssetSelection.ts

```diff
@@ -44,5 +52,19 @@ export const parseAssetSelectionQuery = (
   const tree = parser.start();
 
   const visitor = new AntlrAssetSelectionVisitor(all_assets);
-  return [...visitor.visit(tree)];
+  const all_selection = visitor.visit(tree);
+  const focus_selection = visitor.focus_assets;
+
+  return {
+    all: Array.from(all_selection),
+    focus: Array.from(focus_selection),
+  };
 };
+
+export const filterAssetSelectionByQuery = (
+  all_assets: AssetGraphQueryItem[],
+  query: string,
+): AssetSelectionQueryResult =>
+  featureEnabled(FeatureFlag.flagAssetSelectionSyntax)
+    ? parseAssetSelectionQuery(all_assets, query)
+    : filterByQuery(all_assets, query);
```

Comment:
Don't we need a try/catch here?

Reply:
I think we'll want to fallback to `filterByQuery` if `parseAssetSelectionQuery` throws an error.

Reply:
I think we'll want to fallback to filterByQuery if parseAssetSelectionQuery throws an error. Only because we might launch asset selections in catalog views first, and those selections wouldn't conform to the current syntax, and we wouldn't want to break them... alternatively we could try and migrate them to the new syntax on the frontend when we encounter them 🤔